pkg/volume/git_repo/git_repo_test.go

```diff
@@ -55,106 +56,400 @@ func TestCanSupport(t *testing.T) {
 	}
 }
 
-func testSetUp(plug volume.VolumePlugin, builder volume.Builder, t *testing.T) {
-	var fcmd exec.FakeCmd
-	fcmd = exec.FakeCmd{
-		CombinedOutputScript: []exec.FakeCombinedOutputAction{
-			// git clone
-			func() ([]byte, error) {
-				os.MkdirAll(path.Join(fcmd.Dirs[0], "kubernetes"), 0750)
-				return []byte{}, nil
+// Just a volume with specs
+type Scenario struct {
+	vol               *api.Volume
+	expecteds         []Expected
+	isExpectedFailure bool
+	pluginSpec        PluginSpec
+}
+
+type PluginSpec struct {
+	pluginName string
+	volPath    string
+}
+
+// Expected
+type Expected struct {
+	// The git command
+	cmd []string
+	// The dir of git command is executed
+	dir string
+}
+
+func TestPlugin(t *testing.T) {
+	gitUrl := "https://github.com/GoogleCloudPlatform/kubernetes.git"
+	revision := "2a30ce65c5ab586b98916d83385c5983edd353a1"
+	validPluginSpec := PluginSpec{
+		volPath:    "pods/poduid/volumes/kubernetes.io~git-repo/vol1",
+		pluginName: "kubernetes.io/git-repo",
+	}
+
+	scenarios := map[string]Scenario{
```

Comment:
The style for table tests is to define scenario types inline, like so:

```go
func TestThing(t *testing.T) {
  cases := []struct{
    vol *api.Volume
    isExpectedFailure
  }

  // other stuff
}
```

In this case, go ahead and keep the `Expected` type, but make it called `expectedCommand` (more specific, and non-exported), but move the `Scenario` into an anonymous struct.  Using anonymous structs helps keep the symbol space of a package clean.